rchie01/dotTheory-React
# RCT-103: React Components

# Related Tickets

https://codeandtheory.atlassian.net/browse/RCT-131

# Description

- Creates a simple Scoreboard application using `create-react-app`
- This PR demonstrates the use of building modular interfaces with components, management of state and data flow, lifecycle methods and stateful components, and React Component patterns

# Validation


- Clone the repository using SSH or HTTPS
- The video course challenge is separated into two coding challenges

Part 1:
1. Open the RCT-103 folder and open the _**../create-react-app-demo/scoreboard**_ folder
2. Run `npm start` (tested using _**node v18.0.0**_)
3. Interact with the application by starting/stopping and resetting the timer, remove current players, add new players, and increase and decrease the scores

Part 2:
1. Open the RCT-103 folder and open the _**../scoreboard**_ folder
2. Go Live using the **Live Server** extension in VSCode
5. Interact with the application by increasing and/or decreasing the scores of one of the players

The user should be able to interact with the applications as intended

# Risks
- N/A
